DB/src/main/java/io/deephaven/stream/StreamToTableAdapter.java

```diff
@@ -29,11 +54,83 @@ public StreamToTableAdapter(@NotNull final TableDefinition tableDefinition,
         this.liveTableRegistrar = liveTableRegistrar;
         streamPublisher.register(this);
         liveTableRegistrar.addTable(this);
-        // TODO: Construct table using the supplied definition and a series of chunk-wrapping `SwitchColumnSource`s.
-        table = null;
+
+        columnTypesArray = tableDefinition.getColumnTypesArray();
+        columnNamesArray = tableDefinition.getColumnNamesArray();
+
+        chunkTypes = tableDefinitionToChunkTypes(columnTypesArray);
+        currentChunkSources = makeChunkSources(columnTypesArray, columnNamesArray);
+        nullColumnSources = makeNullColumnSources(currentChunkSources);
+
+        final LinkedHashMap<String, ColumnSource<?>> visibleSources = new LinkedHashMap<>();
+        switchSources = makeSwitchSources(currentChunkSources, visibleSources);
+
+        bufferChunkSources = makeChunkSources(columnTypesArray, columnNamesArray);
```

Comment:
Why do we need this in the constructor? Seems inconsistent with the rest of the code.